runtime/src/bank.rs

```diff
@@ -1216,31 +1216,21 @@ impl Bank {
         vote_account_hashmap: &HashMap<Pubkey, (u64, Account)>,
         rent_to_be_distributed: u64,
     ) {
-        let mut node_stake_hashmap: HashMap<Pubkey, u64> = HashMap::new();
         let mut total_staked = 0;
 
-        vote_account_hashmap
+        let node_stakes = vote_account_hashmap
             .iter()
             .filter_map(|(_vote_pubkey, (staked, account))| {
                 total_staked += *staked;
                 VoteState::deserialize(&account.data)
                     .ok()
-                    .map(|vote_state| (*staked, vote_state.node_pubkey))
+                    .map(|vote_state| (vote_state.node_pubkey, *staked))
             })
-            .collect::<HashMap<u64, Pubkey>>()
-            .drain()
-            .for_each(|(staked, node_pubkey)| {
-                node_stake_hashmap
-                    .entry(node_pubkey)
-                    .and_modify(|stake| {
-                        *stake += staked;
-                    })
-                    .or_insert(staked);
-            });
+            .collect::<HashMap<Pubkey, u64>>();
```

Comment:
why are you collect()ing, then iterating again?

Reply:
nevermind.  you need total_staked, right?

Reply:
yes.

Reply:
you'll want to collect() into something that can handle duplicates, like a vec...

Reply:
@rob-solana Hmm. If vote account and node pubkey has 1:1 relation, then why do we need vec, as there wont be duplicate pubkey.

Reply:
vote and node *should* be 1:1, but might not be...  silly case.  Vec should be faster, too (no hashing)

Reply:
@rob-solana Optimized using vec